engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -76,6 +76,33 @@
   private String recordingNameTopic;
   private long numWindows = Long.MAX_VALUE; // number of windows to record
   private Runnable stopProcedure; // stop procedure to execute
+
+  private static final Logger logger = LoggerFactory.getLogger(TupleRecorder.class);
+
+  // If there are errors processing tuples, don't log an error for every tuple as it could overwhelm the logs.
+  // The property specifies the minumum number of tuples between two consecutive error log statements. Set it to zero to
+  // log every tuple error
+  private static int ERROR_LOG_GAP;
+  int lastLog = -1;
+
+  static {
```

Comment:
That is a general statement. The intent of using a property such as this is that it is a quick way to still allow some customization where we don't yet think or know if it is something we want to expose as regular api to the user, for example as an attribute. We have done this in other places like buffer server. If in future, we start seeing a number of properties like these and we want to handle these better, we could have a common configuration object that will load and provide these properties which will either be accessible via a factory or get passed in, to the individual modules. I prefer not to dwell on this too much right now as the main part of the fix is the other things below.